src/senaite/core/content/worksheet.py

```diff
@@ -0,0 +1,1556 @@
+            default=u"Position"
+        ),
+        required=False,
+        default=1,
+    )
+
+    type = schema.Choice(
+        title=_(
+            u"title_worksheet_layout_analysis_type",
+            default=u"Analysis Type"
+        ),
+        source="senaite.core.vocabularies.analysis_types",
+        required=False,
+        default=u"a",
+    )
+
+    container_uid = UIDReferenceField(
+        title=_(
+            u"title_worksheet_layout_container",
+            default=u"Container"
+        ),
+        allowed_types=("ReferenceSample", "AnalysisRequest",),
+        multi_valued=False,
+        required=False,
+    )
+
+    analysis_uid = UIDReferenceField(
+        title=_(
+            u"title_worksheet_layout_analysis",
+            default=u"Analysis"
+        ),
+        allowed_types=("Analysis", "DuplicateAnalysis", "ReferenceAnalysis",
+                       "RejectAnalysis"),
+        multi_valued=False,
+        required=False,
+    )
+
+
+class IWorksheetSchema(model.Schema):
+    """Worksheet schema interface
+    """
+
+    title = schema.TextLine(
+        title=_(
+            u"title_worksheet_title",
+            default=u"Name"
+        ),
+        required=True,
+    )
+
+    description = schema.Text(
+        title=_(
+            u"title_worksheet_description",
+            default=u"Description"
+        ),
+        required=False,
+    )
+
+    worksheet_template = UIDReferenceField(
+        title=_(
+            u"title_worksheet_worksheettemplate",
+            default=u"Worksheet Template"
+        ),
+        allowed_types=("WorksheetTemplate",),
+        multi_valued=False,
+        required=False,
+    )
+
+    layout_view = DataGridField(
+        title=_(
+            u"title_worksheet_layout",
+            default=u"Layout"
+        ),
+        value_type=DataGridRow(schema=IWorksheetLayoutSchema),
+        default=[],
+        required=False,
+    )
+
+    # all layout info lives in Layout; Analyses is used for back references.
+    analyses = UIDReferenceField(
+        title=_(
+            u"title_worksheet_analyses",
+            default=u"Analyses"
+        ),
+        allowed_types=("Analysis", "DuplicateAnalysis", "ReferenceAnalysis",
+                       "RejectAnalysis"),
+        relationship="WorksheetAnalysis",
+        multi_valued=True,
+        required=True,
+    )
+
+    method = UIDReferenceField(
+        title=_(
+            u"title_worksheet_method",
+            default=u"Method"
+        ),
+        allowed_types=("Method",),
+        vocabulary="senaite.core.vocabularies.available_methods",
+        multi_valued=False,
+        required=False,
+    )
+
+    analyst = schema.TextLine(
+        title=_(
+            u"title_worksheet_analyst",
+            default=u"Analyst"
+        ),
+        required=False,
+        default=u""
+    )
+
+    # TODO Remove. Instruments must be assigned directly to each analysis.
+    instrument = UIDReferenceField(
```

Comment:
You could use this Pull Request as an opportunity to remove the instrument field entirely. The intended behavior is that the instrument selector remains available in the Worksheet view, but when an instrument is selected, the system should assign it individually to each analysis in the worksheet that supports that instrument.